include/seqan/blast/blast_tabular_out.h

```diff
@@ -0,0 +1,799 @@
+            BlastTabular const &)
+{
+    switch (fieldId)
+    {
+        case BlastMatchField<>::Enum::STD:
+             // STD is handled from the calling function
+            break;
+        case BlastMatchField<>::Enum::Q_SEQ_ID:
+            write(s, prefix(match.qId, _firstOcc(match.qId, ' ')));
+            break;
+//         case ENUM::Q_GI: write(s,  * ); break;
+//         case ENUM::Q_ACC: write(s,  * ); break;
+//         case ENUM::Q_ACCVER: write(s,  * ); break;
+        case BlastMatchField<>::Enum::Q_LEN:
+            write(s, match.qLength);
+            break;
+        case BlastMatchField<>::Enum::S_SEQ_ID:
+            write(s, prefix(match.sId, _firstOcc(match.sId, ' ')));
+            break;
+//         case ENUM::S_ALL_SEQ_ID: write(s,  * ); break;
+//         case ENUM::S_GI: write(s,  * ); break;
+//         case ENUM::S_ALL_GI: write(s,  * ); break;
+//         case ENUM::S_ACC: write(s,  * ); break;
+//         case ENUM::S_ACCVER: write(s,  * ); break;
+//         case ENUM::S_ALLACC: write(s,  * ); break;
+        case BlastMatchField<>::Enum::S_LEN:
+            write(s, match.sLength);
+            break;
+        case BlastMatchField<>::Enum::Q_START:
+        {
+            TPos effectiveQStart    = match.qStart;
+            TPos effectiveQEnd      = match.qEnd;
+            _untranslateQPositions(effectiveQStart, effectiveQEnd, match.qFrameShift, match.qLength,
+                                   context.blastProgram, BlastProgramTag<p>());
+            write(s, effectiveQStart);
+        } break;
+        case BlastMatchField<>::Enum::Q_END:
+        {
+            TPos effectiveQStart    = match.qStart;
+            TPos effectiveQEnd      = match.qEnd;
+            _untranslateQPositions(effectiveQStart, effectiveQEnd, match.qFrameShift, match.qLength,
+                                   context.blastProgram, BlastProgramTag<p>());
+            write(s, effectiveQEnd);
+        } break;
+        case BlastMatchField<>::Enum::S_START:
+        {
+            TPos effectiveSStart    = match.sStart;
+            TPos effectiveSEnd      = match.sEnd;
+            _untranslateSPositions(effectiveSStart, effectiveSEnd, match.sFrameShift, match.sLength,
+                                   context.blastProgram, BlastProgramTag<p>());
+            write(s, effectiveSStart);
+        } break;
+        case BlastMatchField<>::Enum::S_END:
+        {
+            TPos effectiveSStart    = match.sStart;
+            TPos effectiveSEnd      = match.sEnd;
+            _untranslateSPositions(effectiveSStart, effectiveSEnd, match.sFrameShift, match.sLength,
+                                   context.blastProgram, BlastProgramTag<p>());
+            write(s, effectiveSEnd);
+        } break;
+//         case ENUM::Q_SEQ: write(s,  * ); break;
+//         case ENUM::S_SEQ: write(s,  * ); break;
+        case BlastMatchField<>::Enum::E_VALUE:
+        {
+            std::string formatString;
```

Comment:
For strings this short this is guaranteed to be storage from the stack; accessing a potentially larger buffer and overwriting it could actually be slower.
But if we have a policy for this I can change it of course.